astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -88,76 +98,95 @@ class galactocentric_frame_defaults(ScienceState):
+                    "galcen_v_sun": [
+                        "https://ui.adsabs.harvard.edu/#abs/2010MNRAS.403.1829S",
+                        "https://ui.adsabs.harvard.edu/#abs/2015ApJS..216...29B",
+                    ],
+                    "z_sun": "https://ui.adsabs.harvard.edu/#abs/2001ApJ...553..184C",
+                    "roll": None,
+                }
+            ),
+        },
+    }
+
     @classmethod
     def get_solar_params_from_string(cls, arg):
         """Return Galactocentric solar parameters given string names for the
         parameter sets.
         """
-
         # Resolve the meaning of 'latest': The latest parameter set is from v4.0
         # - update this as newer parameter choices are added
         if arg == 'latest':
             arg = cls._latest_value
 
-        params = dict()
-        references = dict()
-
-        # Currently, all versions use the same sky position for Sgr A*:
-        params['galcen_coord'] = ICRS(ra=266.4051*u.degree,
-                                      dec=-28.936175*u.degree)
-        references['galcen_coord'] = \
-            'http://adsabs.harvard.edu/abs/2004ApJ...616..872R'
-
-        # The roll angle is the same for both frames:
-        params['roll'] = 0 * u.deg
-
-        if arg == 'pre-v4.0':
-            params['galcen_distance'] = 8.3 * u.kpc
-            references['galcen_distance'] = \
-                'https://ui.adsabs.harvard.edu/#abs/2009ApJ...692.1075G'
-
-            params['galcen_v_sun'] = r.CartesianDifferential([11.1,
-                                                              220+12.24,
-                                                              7.25]*u.km/u.s)
-            references['galcen_v_sun'] = \
-                ['https://ui.adsabs.harvard.edu/#abs/2010MNRAS.403.1829S',
-                 'https://ui.adsabs.harvard.edu/#abs/2015ApJS..216...29B']
-
-            params['z_sun'] = 27.0 * u.pc
-            references['z_sun'] = \
-                'https://ui.adsabs.harvard.edu/#abs/2001ApJ...553..184C'
-
-        elif arg == 'v4.0':
-            params['galcen_distance'] = 8.122 * u.kpc
-            references['galcen_distance'] = \
-                'https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G'
-
-            params['galcen_v_sun'] = r.CartesianDifferential([12.9,
-                                                              245.6,
-                                                              7.78]*u.km/u.s)
-            references['galcen_v_sun'] = \
-                ['https://ui.adsabs.harvard.edu/abs/2018RNAAS...2..210D',
-                 'https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G',
-                 'https://ui.adsabs.harvard.edu/abs/2004ApJ...616..872R']
-
-            params['z_sun'] = 20.8 * u.pc
-            references['z_sun'] = \
-                'https://ui.adsabs.harvard.edu/abs/2019MNRAS.482.1417B'
-
+        if arg in cls._registry:
+            state = cls._registry[arg].copy()
```

Comment:
This also took me bit: the copy is to ensure one cannot just replace, say, "value" and then also change the system state. Correct? Again worth a comment, I think, for our future selves...

Reply:
Yes. I had a few mishaps before I diagnosed that. I put in a comment.